languages/python/sqlalchemy-oso/sqlalchemy_oso/roles.py

```diff
@@ -413,33 +389,16 @@ def add_user_role(session, user, resource, role_name):
     :param role_name: the name of the role to assign to the user
     :type role_name: str
     """
+    # get models
     resource_model = type(resource)
     role_model = get_role_model_for_resource_model(resource_model)
 
-    if role_name not in role_model.choices:
-        raise Exception(
-            f"{role_name} Is not a valid choice for {resource.__class__.__name__} Roles"
-        )
-
-    # try to get role
-    role = (
-        session.query(role_model)
-        .select_from(resource_model)
-        .join(role_model)
-        .filter(resource_model.id == resource.id)
-        .filter(role_model.name == role_name)
-    ).first()
-
-    if role:
-        # TODO: check if user already in role
-        role.users.append(user)
-    else:
-        resource_name = resource_model.__name__.lower()
-        kwargs = {"name": role_name, resource_name: resource, "users": [user]}
-
-        role = role_model(**kwargs)
-        session.add(role)
-        session.commit()
+    # create and save role
+    resource_name = resource_model.__name__.lower()
```

Comment:
These seems like something that would be worth turning into a method so we don't get it wrong somewhere

Reply:
Agreed.